From e3e50f7afd91161142b1850447076d3d60972920 Mon Sep 17 00:00:00 2001
From: Zhou Song <zhous@codeaurora.org>
Date: Tue, 31 Mar 2015 13:53:14 +0800
Subject: [PATCH 79/95] Initialize the start time if not specified in meta data

If the start time is not specified in meta data, the start time
will be assigned with latency of AudioRecord, it's calculated by
allocated frame count which is not the real start time when the
first frame received

Initialize the start time to the current system time if not
specified in meta data

CRs-Fixed: 812379

Change-Id: Ia007ca4592fe1f96e105d4ec48305c578b677bb1
---
 media/libstagefright/AudioSource.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/media/libstagefright/AudioSource.cpp b/media/libstagefright/AudioSource.cpp
index 804f131..7195c91 100644
--- a/media/libstagefright/AudioSource.cpp
+++ b/media/libstagefright/AudioSource.cpp
@@ -115,6 +115,8 @@ status_t AudioSource::start(MetaData *params) {
     int64_t startTimeUs;
     if (params && params->findInt64(kKeyTime, &startTimeUs)) {
         mStartTimeUs = startTimeUs;
+    } else {
+        mStartTimeUs = systemTime() / 1000ll;
     }
     status_t err = mRecord->start();
     if (err == OK) {
-- 
2.5.0

